kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,657 @@
+        InitializeSolutionStepContainer(GetModelPart().MasterSlaveConstraints());
+
+        // Call the builder and solver initialize solution step
+        mpExplicitBuilder->InitializeSolutionStep(mrModelPart);
+    }
+
+    /**
+     * @brief Performs all the required operations that should be done (for each step) after solving the solution step.
+     * @details A member variable should be used as a flag to make sure this function is called only once per step.
+     */
+    virtual void FinalizeSolutionStep()
+    {
+        // FinalizeSolutionStep elements, conditions and constraints
+        FinalizeSolutionStepContainer(GetModelPart().Elements());
+        FinalizeSolutionStepContainer(GetModelPart().Conditions());
+        FinalizeSolutionStepContainer(GetModelPart().MasterSlaveConstraints());
+
+        // Call the builder and solver finalize solution step (the reactions are computed in here)
+        mpExplicitBuilder->FinalizeSolutionStep(mrModelPart);
+    }
+
+    /**
+     * @brief Solves the current step.
+     * The function always return true as convergence is not checked in the explicit framework
+     */
+    virtual bool SolveSolutionStep()
+    {
+        // Call the initialize non-linear iteration
+        InitializeNonLinearIterationContainer(GetModelPart().Elements());
+        InitializeNonLinearIterationContainer(GetModelPart().Conditions());
+        InitializeNonLinearIterationContainer(GetModelPart().MasterSlaveConstraints());
+
+        // Solve the problem assuming that a lumped mass matrix is used
+        SolveWithLumpedMassMatrix();
+
+        // If required, update the mesh with the obtained solution
+        if (mMoveMeshFlag) {
+            MoveMesh();
+        }
+
+        // Call the finalize non-linear iteration
+        FinalizeNonLinearIterationContainer(GetModelPart().Elements());
+        FinalizeNonLinearIterationContainer(GetModelPart().Conditions());
+        FinalizeNonLinearIterationContainer(GetModelPart().MasterSlaveConstraints());
+
+        return true;
+    }
+
+    /**
+     * @brief This sets the level of echo for the solving strategy
+     * @param Level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     */
+    virtual void SetEchoLevel(const int Level)
+    {
+        mEchoLevel = Level;
+    }
+
+    /**
+     * @brief This returns the level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     * @return Level of echo for the solving strategy
+     */
+    virtual int GetEchoLevel()
```

Comment:
needs to be `virtual`?

```suggestion
    virtual int GetEchoLevel() const
```

Reply:
I don't think so. Indeed this should go to the upcoming base strategy.

Reply:
Done.